benchmarks/src/bench.rs

```diff
@@ -69,7 +106,7 @@ pub trait Benchmark: Sized {
         Ok(())
     }
 
-    async fn do_task(&self) -> Result<()>;
+    async fn do_task(&mut self) -> Result<()>;
```

Comment:
My understanding is that the introduction of `mut` here is to support copying the write models in `before_task` rather than `do_task` (which absolutely makes sense).

Personally, I'd rather avoid the `mut` on `do_task` because I think it makes it harder to reason about; how would you feel about, instead, updating the `Benchmark` trait with
```
type TaskState = ();
async fn before_task(&mut self) -> Result<TaskState> {
    Ok(())
}
async fn do_task(&self, state: TaskState) -> Result<()>;
```
That avoids the need for `mut` and the `Option`/`take` ceremony, but might be more type-level complication than it's worth.  WDYT?